src/value.cpp

```diff
@@ -843,7 +843,7 @@ size_t DateValue::copy(byte* buf, ByteOrder /*byteOrder*/) const {
 
   // sprintf wants to add the null terminator, so use oversized buffer
   char temp[9];
-  size_t wrote = static_cast<size_t>(snprintf(temp, sizeof(temp), "%04d%02d%02d", date_.year, date_.month, date_.day));
+  auto wrote = static_cast<size_t>(snprintf(temp, sizeof(temp), "%04d%02d%02d", date_.year, date_.month, date_.day));
   std::memcpy(buf, temp, wrote);
```

Comment:
```suggestion
  auto wrote = snprintf(temp, sizeof(temp), "%04d%02d%02d", date_.year, date_.month, date_.day);
  assert(wrote >= 0);
  std::memcpy(buf, temp, static_cast<size_t>(wrote));
```

Reply:
That's incorrect:

https://en.cppreference.com/w/c/io/fprintf

`snprintf` returns `int`, and it sometimes returns negative numbers on error. I don't think it will ever return a negative number here though, which is why I think an `assert` is sufficient, rather throwing an exception.

Reply:
I also noticed that `snprintf` returns `int` but I thought it would not ever happen in this case. We recently removed most (if not all) the `asserts` from the codebase since those only get executed in Debug builds.

For me the code it is fine as it is at the moment. At the end we need to place 1 static_assert somewhere and I do not see any advantage with having it in the call to `snprintf` compared to doing it later (line 848).

I am approving the PR but not merging it yet. @kevinbackhouse let us know if you really think there is a strong reason/advantage to choose your suggestion.